mesonbuild/options.py

```diff
@@ -509,7 +509,7 @@ def listify(self, value: T.Any) -> T.List[T.Any]:
         except MesonException as e:
             raise MesonException(f'error in option "{self.name}": {e!s}')
 
-    def validate_value(self, value: T.Union[str, T.List[str]]) -> T.List[str]:
+    def validate_value(self, value: T.Any) -> T.List[str]:
```

Comment:
I think it would be better to use `object` here than `Any`. It's not that we don't know the type, it's that we can't really know the type, and object will let mypy yell at us loudly if we start assuming things about the value, and `Any` will ignore them.